radix-engine/src/track/track.rs

```diff
@@ -655,9 +655,11 @@ impl<'s, S: SubstateDatabase, M: DatabaseKeyMapper> SubstateStore for Track<'s,
 
         // Optimization, no need to go into database if the node is just created
         if is_new {
-            return items;
+            return (items, false);
```

Comment:
You might've realised it - a single `boolean` isn't sufficient anyway; during a scan, some of the entries will be from the `Track` and additional entries may/may not be loaded from database.

In the long run, we'll need to change the boolean to a richer data structure to record additional info, e.g. how much is read from track, how much is read from database, any database delete we will have to rebase, and any database write that we will have to charge more.

Reply:
So far I added new type instead of `boolean`, still I need to look into described cases.

Reply:
I extended the data type.